src/utils/index.ts

```diff
@@ -1,196 +1,64 @@
-  const broadcasterID = await getBroadcasterId();
-  const stream = await getBroadcasterStream(broadcasterID);
-
-  const live = stream.data[0] && stream.data[0].type === "live";
-  const id = stream.data[0] && stream.data[0].id;
-
-  return { live, id };
-};
-
-type ChattersData = {
-  broadcaster?: string[];
-  vips?: string[];
-  moderators: string[];
-  staff: string[];
-  admins: string[];
-  global_mods: string[];
-  viewers: string[];
-};
-
-async function getTwitchProfiles(usernames: string[]) {
-  let twitchProfiles: TwitchProfile[] = [];
-
-  usernames = usernames.filter((username: string): boolean => {
-    if (!twitchProfileCache.has(username)) {
-      return true;
-    }
-
-    twitchProfiles.push(twitchProfileCache.get(username));
-    return false;
+import { TwitchProfile } from "../services/twitch/TwitchAPI/apiTypes";
+import {
+  getTwitchChatters,
+  getTwitchProfile
+} from "../services/twitch/TwitchAPI";
+import { default as fetch, RequestInfo } from "node-fetch";
+
+export async function PerformGetRequest<T>(
+  url: RequestInfo,
+  headers: object = {}
+): Promise<T> {
+  const response = await fetch(url, {
+    method: "GET",
+    headers: headers
   });
 
-  const arraysOfUsernames = usernames
-    .reduce(
-      ([_ = [], ...rest], username) =>
-        _.length < 100 ? [[..._, username], ...rest] : [[username], _, ...rest],
-      []
-    )
-    .reverse();
-
-  const userIdQueryStrings = arraysOfUsernames.map(
-    usernames => `login=${usernames.join("&login=")}`
-  );
-
-  for (const query of userIdQueryStrings) {
-    try {
-      const { data: profiles = [] } = await callTwitchApi(
-        `https://api.twitch.tv/helix/users?${query}`,
-        {}
-      );
-
-      profiles.forEach((profile: TwitchProfile) => {
-        twitchProfiles.push(addProfileToCache(profile.login, profile));
-      });
-    } catch (e) {
-      BeastieLogger.warn(`Failed to fetch twitch user ${query}`);
-    }
-  }
-  return twitchProfiles;
+  return await response.json();
 }
 
-const getChatroomViewers = async (): Promise<TwitchProfile[]> => {
-  let chatters: ChattersData;
-  try {
-    chatters = (
-      await callTwitchApi(
-        `https://tmi.twitch.tv/group/user/${config.BROADCASTER_USERNAME}/chatters`,
-        {}
-      )
-    ).chatters;
-  } catch (e) {
-    BeastieLogger.warn(`Failed to get chatroom viewers: ${e}`);
-    return [];
-  }
-
-  return getTwitchProfiles(Object.values(chatters).flat());
-};
-
-export const updateTeammateAwesomeness = async (user, amount) => {
-  const twitchUser = await getTwitchProfile(user);
+export const isGuildMaster = name =>
+  name === config.DISCORD_GUILD_MASTER_USERNAME;
+
+export const updateTeammateAwesomeness = async (
+  user,
+  amount
+): Promise<string> => {
+  let profile: TwitchProfile = await tryCatchLog<TwitchProfile>(
+    async () => getTwitchProfile(user),
+    BeastieLogger.warn,
+    `Failed to get twitch profile for ${user}`
+  );
```

Comment:
```suggestion
  let profile: TwitchProfile = await getTwitchProfile(user) as Promise<TwitchProfile>
    .catch(swallowRejection(`Failed to get twitch profile for ${user}`, BeastieLogger.warn));
```
See ➡️ https://github.com/teamTALIMA/BeastieBot-V3/pull/13/files#r429028595